core/lib/tasks/solidus/split_promotions_with_any_match_policy.rake

```diff
@@ -3,7 +3,8 @@
 namespace :solidus do
   desc "Split Promotions with 'any' match policy"
   task split_promotions_with_any_match_policy: :environment do
-    Spree::Promotion.where(match_policy: :any).includes(:promotion_rules).all.each do |promotion|
+    promotions = Spree::Promotion.where(match_policy: :any)
+    promotions.includes(:promotion_rules, :promotion_actions).find_each do |promotion|
       if promotion.promotion_rules.length <= 1
         promotion.update!(match_policy: :all)
       elsif promotion.active?
```

Comment:
I can't link it, but isn't line 15 unnecessarily nested? I don't think we should be iterating over each action to create a duplicate for every rule 🤔 
```ruby
new_promotion.promotion_actions = promotion.actions.map do |action|
```
If we want to make it faster, we could move the iteration out.

Reply:
This is about creating a setup where promotions with a `match_policy` of `any` are transformed into multiple promotions, one per rule of the existing promotions. It might look like a lot of data, but that's really what `match_policy: any` means...

Reply:
Oh of course! 🤦  Thanks for your reply, forgive my mind going blank on this 😅